assets/js/modules/analytics/datastore/properties.js

```diff
@@ -179,21 +179,26 @@ const baseActions = {
 
 			registry.dispatch( STORE_NAME ).setInternalWebPropertyID( internalPropertyID || '' );
 
-			// Clear any profile ID selection in the case that selection falls to the getProfiles resolver.
-			registry.dispatch( STORE_NAME ).setProfileID( '' );
+			const existingProfileID = registry.select( STORE_NAME ).getProfileID(); // eslint-disable-line @wordpress/no-unused-vars-before-return
+			const profiles = yield Data.commonActions.await( registry.__experimentalResolveSelect( STORE_NAME ).getProfiles( accountID, propertyID ) );
```

Comment:
This is a bit long for one line 🙂 
```suggestion
			const profiles = yield Data.commonActions.await(
				registry.__experimentalResolveSelect( STORE_NAME ).getProfiles( accountID, propertyID )
			);
```